### ISSUE TITLE
Import should scale based on height only

### ISSUE THREAD

**Issue by mwoehlke-kitware** [fontforge/fontforge#4488]
I am having roughly the same issue as #3116. Unfortunately, that was closed without being resolved.

I am using *the Python API* to generate a font. One glyph is wider than it is tall. When I import it, it gets shrunk, resulting in a glyph that is misaligned.

However, I may also need to import glyphs that *should* be scaled. Possibly I can compute the scale factor myself (how?), but in general, I think FF should only use the *height* to scale, rather than squishing wide glyphs to fit into a square box. (The only time I can see this making sense is if dealing with glyphs for a vertical writing system. In all other cases, this results in the vertical position of the glyph having no relation to the SVG viewbox, resulting in a glyph that is virtually guaranteed to have incorrect positioning.)

I am using fontforge-20200314-5.fc32.x86_64 (Fedora).

### Steps to reproduce the behavior
- Create an SVG that is wider than it is tall and has SVG height == target glyph height.
- Import the SVG into a font.

The SVG should not be scaled. However, currently it is scaled by `glyph_size / actual_width`, resulting in the glyph being pushed upwards and left of where it should be.

### Possible solution/fix/workaround

As noted, only look at the *vertical* size when computing scaling.

**Comment by skef** [comment]
I agree that the current default heuristic is a bit strange but we might have to do some research before (in effect) changing the API.

However, do you *need* the vertical scaling? If you don't just add `scale=False` to your call to `glyph.importOutlines()` and your problem should go away. (This was added in the release last spring.)

**Comment by mwoehlke-kitware** [comment]
I would prefer to either have automatic vertical scaling, or the ability to compute my own scale factor. My input SVGs aren't necessarily "correct scale" to start with. I can fix that, but it's extra work. (For now that's the work-around I'm forced to use...)

**Comment by skef** [comment]
Does anyone know why GW might have done this with the width in 2007? @frank-trampe ?

```
        if ( width>height ) {
            if ( scale && swidth!=0 ) {
                st.transform[0] *= em_size/swidth;
                st.transform[3] *= em_size/swidth;
            }
        } else {
            if ( scale && sheight!=0 ) {
                st.transform[0] *= em_size/sheight;
                st.transform[3] *= em_size/sheight;
            }
        }
```

(I added `scale` to the guards)

**Comment by skef** [comment]
@frank-trampe Should we just not do the width adjustment?

**Comment by rgpublic** [comment]
> However, do you _need_ the vertical scaling? If you don't just add `scale=False` to your call to `glyph.importOutlines()` and your problem should go away. (This was added in the release last spring.)

I wonder how I can call that option from Python. No matter what I try it fails. I'm using the script from https://github.com/pteromys/svgs2ttf

IMPORT_OPTIONS = ('removeoverlap', 'correctdir', 'scale=False')
g.importOutlines(src, IMPORT_OPTIONS)

doesn't work (Unknown PostScript import flag). If I check the documentation here: 

https://fontforge.org/docs/scripting/python/fontforge.html#glyph

I don't even see the option "removeoverlap" mentioned. And now example is give on how I can specify values - just a list of keywords. Very confusing. Hope someone can give me a little hint.

**Comment by skef** [comment]
I'm not sure why `removeoverlap is being passed this way. You can first just try:

```
g.importOutlines(src, correctdir=False, scale=False)
```

**Comment by rgpublic** [comment]
@skef: Your statement is accepted, thanks. Still I wonder why the original IMPORT_OPTIONS=... worked with the ()-tuple. Shouldn't there be a need to unpack the tuple to keyword arguments with "*"? But, then again, I don't know that much about Python...

Next problem: If I specify "scale=False" the font isn't accepted by Firefox anymore. All very weird, but I've settled on adjusting the icons manually for now. Thanks again!

**Comment by Neptilo** [comment]
To get back to the original issue, are there plans to remove that `if ( width>height )` conditional?

**Comment by skef** [comment]
@Neptilo sorry, the [project leadership](https://github.com/orgs/fontforge/people?query=role%3Aowner) doesn't spend much time in the issue or PR queues anymore, making it difficult to judge in advance whether a PR is worthwhile or will wind up being merged. Until that changes or there's a fork I'm not sure what can be done to solve this kind of issue.

**Comment by ctrlcctrlv** [comment]
put @skef in charge of merges 2021

#MakeFontForgeGreatAgain
